arch/arc/include/v2/irq.h

```diff
@@ -52,8 +52,12 @@ static ALWAYS_INLINE void _irq_setup(void)
 	);
 
 	k_cpu_sleep_mode = _ARC_V2_WAKE_IRQ_LEVEL;
+#ifdef CONFIG_ARC_NORMAL_FIRMWARE
+	/* \todo use sjli instruction */
```

Comment:
not sure an sjli call would be required here, this setup is only done once and could be done by Secure firmware before switching to Normal Zephyr.